src/param.rs

```diff
@@ -970,28 +988,24 @@ impl AudioParamProcessor {
         let clamped = self.intrisic_value.clamp(self.min_value, self.max_value);
         self.current_value.store(clamped);
 
-        // no even in timeline and buffer is clean, we can just reuse current buffer,
-        // instead of clearing to just refill with same values
-        if self.event_timeline.is_empty() && self.is_buffer_clean {
-            return;
-        }
-
-        // Define if intrisic buffer can be considered be clean in next call.
-        // If the timeline is empty, we know the buffer will be filled with a
-        // constant value. @note: there could other valid cases
-        self.is_buffer_clean = self.event_timeline.is_empty();
-
-        // populate the buffer for this block
+        // clear the buffer for this block
         self.buffer.clear();
 
-        let next_block_time = dt.mul_add(count as f64, block_time);
         let is_a_rate = self.is_a_rate.load(Ordering::SeqCst);
         let is_k_rate = !is_a_rate;
+        let is_timeline_empty = self.event_timeline.is_empty();
 
-        if is_k_rate {
-            self.buffer.resize(count, self.intrisic_value);
+        if is_k_rate || is_timeline_empty {
+            self.buffer.resize(1, self.intrisic_value);
```

Comment:
`self.buffer.push(self.intrisic_value)` is a bit more intuitive maybe

Reply:
yup done